src/Bundle/Extractor/Property.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace Zenstruck\Foundry\Bundle\Extractor;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Exception;
+use ReflectionClass;
+
+class Property
+{
+    /**
+     * @var array
+     */
+    private $properties = [];
+
+    /**
+     * @var EntityManagerInterface
+     */
+    private $em;
+
+    public function __construct(EntityManagerInterface $em)
+    {
+        $this->em = $em;
+    }
+
+    /**
+     * @TODO Support for relations is missing
+     */
+    public function getFakerMethodFromDoctrineFieldMappings(ReflectionClass $entity): array
+    {
+        $classMetaData = $this->em->getClassMetadata($entity->getName());
+        $identifierFieldNames = $classMetaData->getIdentifierFieldNames();
+
+        foreach ($classMetaData->fieldMappings as $property) {
+            // IGNORE FIELD IF IDENTIFIER
+            if (\in_array($property['fieldName'], $identifierFieldNames)) {
+                continue;
+            }
+
+            // CREATE FROM DOCTRINE TYPE IF PROP IS NOT NULLABLE
+            if (!$property['nullable']) {
+                $this->properties[$property['fieldName']] = $this->createFakerMethodFromDoctrineType($property['type']);
+            }
+        }
+
+        return $this->properties;
+    }
+
+    /**
+     * @throws Exception
+     */
+    public function createFakerMethodFromDoctrineType(string $doctrineType): string
+    {
+        $doctrineType = \mb_strtoupper($doctrineType);
+
+        if (\array_key_exists($doctrineType, DoctrineTypes::DOCTRINE_TYPES)) {
+            return DoctrineTypes::DOCTRINE_TYPES[$doctrineType];
+        }
+
+        return 'null, // @TODO add '.$doctrineType.' manually';
+    }
+
+    /**
+     * @TODO get relations which cant be NULL
+     */
+    public function getPropertiesFromDoctrineRelations()
```

Comment:
What do you think about, for this PR, just adding the code to detect non-nullable relations and adding a todo to the user's factory (just like we do with unsupported types)?

Reply:
if we merge this PR first we can test next CI checks run automatic? And yes add todo for not nullable relations is a good next todo. Also some naming refactoring.

Reply:
> if we merge this PR first we can test next CI checks run automatic

I think so, but I'm not positive. Want me to merge as is and you can add in a follow-up PR?

Reply:
yes would be great 👍